client/engine/chainservice/chainservice.go

```diff
@@ -16,35 +16,46 @@ type Event interface {
 	ChannelID() types.Destination
 }
 
-// CommonEvent declares fields shared by all chain events
-type CommonEvent struct {
+// commonEvent declares fields shared by all chain events
+type commonEvent struct {
 	channelID types.Destination
 	BlockNum  uint64
 }
 
-func (ce CommonEvent) ChannelID() types.Destination {
+func (ce commonEvent) ChannelID() types.Destination {
 	return ce.channelID
 }
 
+type asset struct {
+	AssetAddress common.Address
+	AssetAmount  *big.Int
+}
```

Comment:
Nit: Elsewhere we have used `asset` as a synonym for `assetAddress`...

Might this be better?
```go
type assetAmount struct {
    Asset common.Address
    Amount *big.Int
}
```
(or even `assetAndAmount`)

Also, I'm unsure about how much cleaner it is to bundle these fields into a struct. I think you've saved most lines you saved were due to using a constructor (which anyway accepts the two fields separately)?